packages/selected-network-controller/tests/SelectedNetworkController.test.ts

```diff
@@ -73,56 +104,27 @@ describe('SelectedNetworkController', () => {
     });
   });
 
-  it('updates the networkClientId for the metamask domain when the networkControllers selectedNetworkClientId changes', () => {
-    const messenger = buildMessenger();
-    const options: SelectedNetworkControllerOptions = {
-      messenger: buildSelectedNetworkControllerMessenger(messenger),
-    };
-    const controller = new SelectedNetworkController(options);
-    controller.setNetworkClientIdForMetamask('oldNetwork');
-    expect(controller.state.domains.metamask).toBe('oldNetwork');
-
-    const patch = [
-      {
-        path: ['selectedNetworkClientId'],
-        op: 'replace' as const,
-        value: 'newNetwork',
-      },
-    ];
-
-    const state = {
-      ...networkControllerDefaultState,
-      selectedNetworkClientId: 'newNetwork',
-    };
-    messenger.publish('NetworkController:stateChange', state, patch);
-    expect(controller.state.domains.metamask).toBe('newNetwork');
-  });
-
-  it("does not update the state if the network controller state changes but the selected network hasn't", () => {
-    const mockMessagingSystem = {
-      registerActionHandler: jest.fn(),
-      subscribe: jest.fn(),
-      publish: () => jest.fn(),
-    };
-    const options: SelectedNetworkControllerOptions = {
-      messenger: mockMessagingSystem as any,
-    };
-    const controller = new SelectedNetworkController(options);
-    controller.setNetworkClientIdForMetamask('oldNetwork');
-    expect(controller.state.domains.metamask).toBe('oldNetwork');
+  describe('getProviderAndBlockTracker', () => {
+    it('returns a proxy provider and block tracker when there is one already', () => {
+      const options: SelectedNetworkControllerOptions = {
+        messenger: buildSelectedNetworkControllerMessenger(),
+      };
+      const controller = new SelectedNetworkController(options);
+      controller.setNetworkClientIdForDomain('example.com', 'network7');
+      const result = controller.getProviderAndBlockTracker('example.com');
+      expect(result).toBeDefined();
+    });
 
-    const stateChangeHandler = mockMessagingSystem.subscribe.mock.calls[0][1];
-    const state: any = {
-      selectedNetworkClientId: 'newNetwork',
-    };
-    const patch = [
-      {
-        path: ['anythingelse'],
-        op: 'replace',
-        value: 'abc',
-      },
-    ];
-    stateChangeHandler(state, patch);
-    expect(controller.state.domains.metamask).toBe('oldNetwork');
+    it('creates a new proxy provider and block tracker when there isnt one already', () => {
+      const options: SelectedNetworkControllerOptions = {
+        messenger: buildSelectedNetworkControllerMessenger(),
+      };
+      const controller = new SelectedNetworkController(options);
+      expect(
+        controller.getNetworkClientIdForDomain('test.com'),
```

Comment:
Do we need to know that this method does the right thing in this test?

Reply:
it asserts that there is not a false positive on the test, by making sure that the networkClientId is not set for the domain 'test.com'.